src/test/java/com/youengineering/openapi/GenerateTest.java

```diff
@@ -69,16 +70,52 @@ private GenerateMojo createGenerateMojo(Framework framework) throws Exception {
     private void assetGeneratedFile_Json(Framework framework) throws Exception {
         JsonNode actualNode = Json.mapper().readTree(getOpenAPIFile(framework, "generated", OutputFormat.json));
         JsonNode expectedNode = Json.mapper().readTree(getOpenAPIFile(framework, "expected", OutputFormat.json));
-        assertEquals(expectedNode, actualNode);
+
+        // Removing all null attributes as openApi deserializer user node.get("extensions") to get value.
+        // And if any of the vales are null, it is getting treated as "null" (String type) and failing with class cast exception
+        removeNullsFromJsonNode(actualNode);
+        removeNullsFromJsonNode(expectedNode);
+        OpenAPI actualOpenAPI = openAPIV3Parser.parseJsonNode("", actualNode).getOpenAPI();
+        OpenAPI expectedOpenAPI = openAPIV3Parser.parseJsonNode("", expectedNode).getOpenAPI();
+        assertOpenApis(actualOpenAPI, expectedOpenAPI);
     }
 
     private void assetGeneratedFile_Yaml(Framework framework) {
-        OpenAPI actualOpenAPI = openAPIV3Parser.read(getOpenAPIFile(framework, "generated", OutputFormat.yaml).getPath());
-        OpenAPI expectedOpenAPI = openAPIV3Parser.read(getOpenAPIFile(framework, "expected", OutputFormat.yaml).getPath());
+        OutputFormat yaml = OutputFormat.yaml;
+        OpenAPI actualOpenAPI = openAPIV3Parser.read(getOpenAPIFile(framework, "generated", yaml).getPath());
+        OpenAPI expectedOpenAPI = openAPIV3Parser.read(getOpenAPIFile(framework, "expected", yaml).getPath());
+        assertOpenApis(actualOpenAPI, expectedOpenAPI);
+    }
+
+    private static void assertOpenApis(OpenAPI actualOpenAPI, OpenAPI expectedOpenAPI) {
+        assertNotNull(actualOpenAPI);
+        assertEquals(expectedOpenAPI.getOpenapi(), actualOpenAPI.getOpenapi());
+        assertEquals(expectedOpenAPI.getInfo(), actualOpenAPI.getInfo());
+        assertEquals(expectedOpenAPI.getExternalDocs(), actualOpenAPI.getExternalDocs());
+        assertEquals(expectedOpenAPI.getServers(), actualOpenAPI.getServers());
+        assertEquals(expectedOpenAPI.getSecurity(), actualOpenAPI.getSecurity());
+        assertEquals(expectedOpenAPI.getTags(), actualOpenAPI.getTags());
+        assertEquals(expectedOpenAPI.getComponents(), actualOpenAPI.getComponents());
+        assertEquals(expectedOpenAPI.getExtensions(), actualOpenAPI.getExtensions());
+        assertEquals(expectedOpenAPI.getJsonSchemaDialect(), actualOpenAPI.getJsonSchemaDialect());
+        assertEquals(expectedOpenAPI.getSpecVersion(), actualOpenAPI.getSpecVersion());
+        assertEquals(expectedOpenAPI.getWebhooks(), actualOpenAPI.getWebhooks());
+        assertEquals(expectedOpenAPI.getPaths(), actualOpenAPI.getPaths());
```

Comment:
You don't need these, the OpenAPI classes implement equals and hashCode.

![image](https://user-images.githubusercontent.com/29703834/184361882-41550676-ba17-40f3-b257-9de07f24d347.png)

Reply:
yes. But I just added individual assertions because, If we rely on the whole object asserts then it becomes very tricky to find which exact part of the generated swagger is not matching.
Just added individual assertion as well, so that it can quickly explain which part of the swagger is not matching.